### ISSUE TITLE
Keras-LSTM Multiple Class Classification Problem

### ISSUE THREAD

**Issue by xisnu** [fchollet/keras#1010]
I am trying to use LSTM for a simple classification problem with IRIS data set. My training input is an array [150,4] 150 samples each with 4 features and my target is [150,3], as number of classes is 3. I am using the same set for testing. So I have created the following network

```
model = Sequential()
model.add(Embedding(4,64))
model.add(LSTM(64,activation='tanh', inner_activation='hard_sigmoid'))#Create Input Layer
model.add(Dense(3, init='uniform'))#Create output layer
model.add(Activation('softmax')) 
```

the network is successfully created. Now I am training with the following parameters.

```
model.fit(inputs, targets,nb_epoch=100,batch_size=10,shuffle=True,verbose=1,show_accuracy=True)
```

**inputs** is the [150,4] array and **targets** is the [150,3] array as said previously. The class labels are binary i.e 1,0,0 represents **class A**, 0,1,0 represents **class B** and so on. Now I have used SGD and RMSProp for experiments. But the system is always identifying a particular class. So the accuracy is always around 33.33%. Moreover all the 3 outputs for each of the 150 samples are producing same results. i.e I am getting 150 rows of same data when testing.

What am I doing wrong. May be I am misunderstanding the use of LSTM in Keras completely. I have experimented with the same data in MLP. It gave excellent result. I am just curious in LSTM.

Any suggestion is highly appreciated. Thank you for your time.

**Comment by xisnu** [comment]
I was watching Issue #274

**Comment by lukedeo** [comment]
LSTMs are for sequential data, i.e., when you have a `timestep` dimension, not for fixed representations with no sequential component.

**Comment by xisnu** [comment]
Thank you @lukedeo , I was guessing the same. But I came by few papers where people used LSTM for handwriting recognition. That is why I was trying to fit my model in LSTM

**Comment by viksit** [comment]
@xisnu which paper are you referring to?

**Comment by tursunwali** [comment]
@xisnu 
you said
"May be I am misunderstanding the use of LSTM in Keras completely. I have experimented with the same data in MLP. It gave excellent result. I am just curious in LSTM."
could you share that code with me ! 
I have used MLP in weka , it gives good result as well.
I wonder what keras code looks like ?

**Comment by xisnu** [comment]
@tursunwali I have no idea about WEKA. But I believe Keras would let you customize your model with much more flexibility. You can try this example [https://www.kaggle.com/kernels/diff/2570/4790](url) for a MLP.
